source/common/filesystem/posix/directory_iterator_impl.cc

```diff
@@ -42,39 +42,39 @@ void DirectoryIteratorImpl::nextEntry() {
   }
 
   if (entry == nullptr) {
-    entry_ = {"", FileType::Other};
+    entry_ = {"", FileType::Other, 0};
   } else {
     const std::string current_path(entry->d_name);
     const std::string full_path(directory_path_ + "/" + current_path);
-    entry_ = {current_path, fileType(full_path, os_sys_calls_)};
+    entry_ = makeEntry(entry->d_name);
```

Comment:
since you are computing full_path in makeEntry you can drop the 2 temps above.

Reply:
Done.